core/src/main/java/io/grpc/CallCredentials.java

```diff
@@ -35,83 +34,51 @@
  * pass that reference to gRPC for usage. However, code may not call or implement the {@code
  * CallCredentials} itself if it wishes to only use stable APIs.
  */
-public interface CallCredentials {
-  /**
-   * The security level of the transport. It is guaranteed to be present in the {@code attrs} passed
-   * to {@link #applyRequestMetadata}. It is by default {@link SecurityLevel#NONE} but can be
-   * overridden by the transport.
-   *
-   * @deprecated transport implementations should use {@code
-   * io.grpc.internal.GrpcAttributes.ATTR_SECURITY_LEVEL} instead.
-   */
-  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1914")
-  @Grpc.TransportAttr
-  @Deprecated
-  public static final Key<SecurityLevel> ATTR_SECURITY_LEVEL =
-      Key.create("io.grpc.internal.GrpcAttributes.securityLevel");
-
-  /**
-   * The authority string used to authenticate the server. Usually it's the server's host name. It
-   * is guaranteed to be present in the {@code attrs} passed to {@link #applyRequestMetadata}. It is
-   * by default from the channel, but can be overridden by the transport and {@link
-   * io.grpc.CallOptions} with increasing precedence.
-   */
-  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1914")
-  @Grpc.TransportAttr
-  @Deprecated
-  public static final Key<String> ATTR_AUTHORITY = Key.create("io.grpc.CallCredentials.authority");
+public abstract class CallCredentials {
```

Comment:
The comments starting on line 31 should be changed to reflect this is no longer an interface but an abstract class.